extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,105 @@
+import os.path
+
+from packageurl import PackageURL
+from cyclonedx.factory.license import LicenseFactory
+from cyclonedx.model import ExternalReference, ExternalReferenceType, LicenseChoice, XsUri
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component, ComponentType
+from cyclonedx.output.json import JsonV1Dot4
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+from conans.client.graph.graph import Node
+
+lFac = LicenseFactory()
+
+
+def package_type_to_component_type(pt: str) -> ComponentType:
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if ids is None:
+        return None
+    if not isinstance(ids, tuple):
+        ids = [ids]
+    return [LicenseChoice(license=lFac.make_from_string(i)) for i in ids]
+
+
+def package_url(node: Node) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    return PackageURL(
+        type="conan",
+        name=node.conanfile.name,
+        version=node.conanfile.version,
+        qualifiers={
+            "prev": node.prev,
+            "rref": node.ref.revision,
```

Comment:
@hedtke  I think you need to check a couple of None values.
For instance the following conanfile.txt will lead to
AttributeError: 'NoneType' object has no attribute 'revision'
```
[requires]
openssl/1.1.1t
```

Reply:
None value checked and test added